base/collection.go

```diff
@@ -621,14 +624,14 @@ func (b *GocbV2Bucket) DefaultDataStore() sgbucket.DataStore {
 }
 
 // NamedDataStore returns a collection on a bucket within the given scope and collection.
-func (b *GocbV2Bucket) NamedDataStore(name sgbucket.DataStoreName) sgbucket.DataStore {
+func (b *GocbV2Bucket) NamedDataStore(name sgbucket.DataStoreName) (sgbucket.DataStore, error) {
 	c, err := NewCollection(
 		b,
 		b.bucket.Scope(name.ScopeName()).Collection(name.CollectionName()))
 	if err != nil {
-		panic(err)
+		return nil, fmt.Errorf("attempting to create/update database with a scope/collection that is not found: %w", ErrAuthError)
```

Comment:
This method may not always be called in the context of creating/updating a database.
I think we should keep the message specific to collection not found and the caller of this method can decide what extra context to add to the message if necessary.

We're also losing the actual contents of the underlying error `err`, here because we wanted the base error to return the ErrAuthError type for uniform error handling for non-existent/inaccessible bucket.

What happens if `err` is non-nil but not because the collection was not found? Should we be handling the not-found case specifically to return the AuthError and return `err` for everything else?